sitestats/sitestats-impl/src/java/org/sakaiproject/sitestats/impl/report/ReportManagerImpl.java

```diff
@@ -318,12 +318,26 @@ private ReportProcessedParams processReportParams(ReportParams params, boolean r
 			if(params.isWhatLimitedResourceIds() && params.getWhatResourceIds() != null){
 				rpp.resourceIds = new ArrayList<String>();
 				Iterator<String> iR = params.getWhatResourceIds().iterator();
-				while (iR.hasNext())
-					rpp.resourceIds.add("/content" + iR.next());
+				while (iR.hasNext()) {
+					String next = iR.next();
+					rpp.resourceIds.add("/content" + next);
+				}
 			}
 			if(params.isWhatLimitedAction() && params.getWhatResourceAction() != null) {
 				rpp.resourceAction = params.getWhatResourceAction();			
 			}
+		} else if (params.getWhat().equals(ReportManager.WHAT_LESSONPAGES)){
+			rpp.resourceIds = null;
+			rpp.resourceAction = null;
+			if (params.isWhatLimitedResourceIds() && params.getWhatResourceIds() != null) {
+				rpp.resourceIds = new ArrayList<String>();
+				for (String iR : params.getWhatResourceIds()) {
+					rpp.resourceIds.add("/lessonbuilder" + iR);
+                }
```

Comment:
The indentation was fine. You can't see the matching if - it's above this block.